directus-extension-charging-stations-bundle/package.json

```diff
@@ -0,0 +1,50 @@
+{
+	"name": "directus-extension-charging-stations-bundle",
+	"description": "Directus extension bundle for CitrineOS to support charging stations",
+	"icon": "extension",
+	"version": "1.0.0",
+	"keywords": [
+		"directus",
+		"directus-extension",
+		"directus-extension-bundle"
+	],
+	"type": "module",
+	"files": [
+		"dist"
+	],
+	"directus:extension": {
+		"type": "bundle",
+		"path": {
+			"app": "dist/app.js",
+			"api": "dist/api.js"
+		},
+		"entries": [{
+			"type": "display",
+			"name": "Present or True Count",
+			"source": "src/display-true-count/index.ts"
+		}, {
+			"type": "endpoint",
+			"name": "Charging Stations endpoints",
+			"source": "src/endpoints-charging-stations/index.ts"
+		}, {
+			"type": "hook",
+			"name": "On Create Charging Station",
+			"source": "src/hook-on-create-charging-station/index.ts"
+		}],
+		"host": "^10.10.0"
+	},
+	"scripts": {
+		"build": "directus-extension build",
```

Comment:
more common variation for other extensions that may not require the 2 app/api separations: ```"scripts": {
		"build": "directus-extension build",
		"build-dev": "directus-extension build --no-minify --sourcemap -t interface -i src/index.ts -o ../../../../extensions/interfaces/name/index.js",
		"dev": "directus-extension build --watch --no-minify --sourcemap -t interface -i src/index.ts -o ../../../../extensions/interfaces/name/index.js",
		"link": "directus-extension link"
	}``` with the the directus extensions settings: ```"directus:extension": {
		"type": "interface",
		"path": "../../../../extensions/interfaces/name/index.js",
		"source": "src/index.ts",
		"host": "10...."
	},```

Reply:
> this also ensures that when you run locally it will include a sourcemap and a watcher while also compiling it to the correct extensions directory

This would be quite nice!
If I'm understanding you correctly, the only way to support this would be to break out the bundle into 3 separate extensions, then use the `directus:extension.path` param to place the built code directly into the file structure used by Directus. This wouldn't be able to build to both `Server` and `Swarm`, correct? Are there other ways to accomplish this?

Reply:
No this would be only for Directus. Looking at your PR without even considering the logic that you are trying to achieve, I think I have some points in terms of project structure and sharing code between the extensions as well as building and debugging that I would love to run by you!